### PR TITLE
fix #3498 - correctly consider marks on unittest classes

### LINKED ISSUE #3498: 3.6.0 breaks usefixtures when using unittest/unittest2
Thanks for submitting an issue!

Here's a quick checklist in what to include:

- [x] Include a detailed description of the bug or suggestion
`@pytest.mark.usefixtures()` on test cases in classes that inherit from unittest.TestCase and unittest2.TestCase no longer works.  The fixtures are completely ignored.

Given your README.rst says `Can run unittest (or trial), nose test suites out of the box;`, I'm assuming this is an appropriate bug to file.

- [x] `pip list` of the virtual environment you are using
In both venvs, I installed pytest and unittest2.  The remaining packages are just dependencies

python 3.6.3
```
Owens-MacBook-Pro:bug owen$ pip list
Package        Version
-------------- -------
atomicwrites   1.1.5
attrs          18.1.0
linecache2     1.0.0
more-itertools 4.1.0
pip            10.0.1
pluggy         0.6.0
py             1.5.3
pytest         3.6.0
setuptools     39.2.0
six            1.11.0
traceback2     1.4.0
unittest2      1.1.0
wheel          0.31.1
```

python 2.7.10
```
Owens-MacBook-Pro:bug owen$ pip list
Package        Version
-------------- -------
atomicwrites   1.1.5
attrs          18.1.0
funcsigs       1.0.2
linecache2     1.0.0
more-itertools 4.1.0
pip            10.0.1
pluggy         0.6.0
py             1.5.3
pytest         3.6.0
setuptools     39.2.0
six            1.11.0
traceback2     1.4.0
unittest2      1.1.0
wheel          0.31.1
```
- [x] pytest and operating system versions
pytest 3.6.0 
macOS High Sierra 10.13.4

- [x] Minimal example if possible

conftest.py
```python
import pytest


@pytest.fixture(scope='function')
def fixture1():
    print("in fixture1")
    yield


@pytest.fixture(scope='function')
def fixture2():
    print("in fixture2")
    yield
```

tests.py
```python
import pytest
# import unittest
# import unittest2


# class Tests(unittest.TestCase):
# class Tests(unittest2.TestCase):
class Tests(object):

    @pytest.mark.usefixtures("fixture1")
    def test_one(self):
        print("inside test_one")

    @pytest.mark.usefixtures("fixture1", "fixture2")
    def test_two(self):
        print("inside test_two")
```

Results identical between python 2.7.10 and 3.6.3:

using `class Tests(object)`
```
Owens-MacBook-Pro:bug owen$ pytest -s tests.py
=============================== test session starts ===============================
platform darwin -- Python 2.7.10, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /Users/owen/repos/sandbox/bug, inifile:
collected 2 items

tests.py in fixture1
inside test_one
.in fixture1
in fixture2
inside test_two
.

============================ 2 passed in 0.02 seconds =============================
```

using `class Tests(unittest.TestCase)`
```
Owens-MacBook-Pro:bug owen$ pytest -s tests.py
=============================== test session starts ===============================
platform darwin -- Python 2.7.10, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /Users/owen/repos/sandbox/bug, inifile:
collected 2 items

tests.py inside test_one
.inside test_two
.

============================ 2 passed in 0.01 seconds =============================
```
Notice `in fixture1` and `in fixture2` is not printed

using `class Tests(unittest2.TestCase)`
```
Owens-MacBook-Pro:bug owen$ pytest -s tests.py
=============================== test session starts ===============================
platform darwin -- Python 2.7.10, pytest-3.6.0, py-1.5.3, pluggy-0.6.0
rootdir: /Users/owen/repos/sandbox/bug, inifile:
collected 2 items

tests.py inside test_one
.inside test_two
.

============================ 2 passed in 0.04 seconds =============================
```
Notice `in fixture1` and `in fixture2` is not printed

I also installed pytest 3.5.1 instead of 3.6.0 to show it was working before using unittest2:
```
Owens-MacBook-Pro:bug owen$ pytest -s tests.py
=============================== test session starts ===============================
platform darwin -- Python 2.7.10, pytest-3.5.1, py-1.5.3, pluggy-0.6.0
rootdir: /Users/owen/repos/sandbox/bug, inifile:
collected 2 items

tests.py running fixture1
inside test_one
.running fixture1
running fixture2
inside test_two
.

============================ 2 passed in 0.02 seconds =============================
```

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs (you can delete this text from the final description, this is
just a guideline):

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](/changelog/README.rst) for details.
- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [x] Target the `features` branch for new features and removals/deprecations.
- [x] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

File: testing/test_unittest.py [hunk 1]
```diff
@@ -828,3 +828,64 @@ def test_should_not_run(self):
     """)
     reprec = testdir.inline_run()
     reprec.assertoutcome(passed=1)
+
+
+@pytest.mark.issue(3498)
+@pytest.mark.parametrize("base", [
+    'six.moves.builtins.object',
+    'unittest.TestCase',
+    'unittest2.TestCase',
```
Comment (nicoddemus) [inline testing/test_unittest.py]: We never install `unittest2`, should we add it to our testing dependencies in `tox.ini`?

Comment (RonnyPfannschmidt) [reply testing/test_unittest.py]: atm i dont think thats necessary

**Comment by owenleonard** [comment]
I pulled down the changes and verified they work correctly for my use cases.  Thanks for such a quick response!

**Comment by nicoddemus** [comment]
Thanks @RonnyPfannschmidt!

We should have a `3.6.1` release soon

**Pull request merged**